### ISSUE TITLE
3g from wifi warning request

### ISSUE THREAD

**Issue by ahmedre** [quran/quran_android#498]
we got some feedback that someone was downloading quran over wifi, then lost wifi so it continued downloading over 3g. he was wondering if we can have an option or something.

we could have a flag in preferences ("only download on wifi," enabled by default). we could keep the same question when you try to download a sura "on 3g, download?" - but if we're on wifi and the state changes, we can just stop the download or pause it until connection returns.

even if we don't do this now, we should keep it in mind if we do any more download manager refactoring so that something like this becomes easier to implement in the future.

**Comment by murtraja** [comment]
If I understand this correctly, then the task at hand involves the following
  1. Have a flag in preferences: `Only Download on Wifi` by default true
  2. If user requests download on Data, we remind him/her and then proceed as per his/her choice irrespective of flag
  3. If user is currently downloading on Wifi, and the state changes, we stop/pause/schedule the download untill the connection state change to Wifi

Can you let me know if this is still relevant, I would like to work on this.

**Comment by ahmedre** [comment]
yes, that's right. also please keep in mind there's a check today (only at the start of the download). some people are on paid plans so don't want to pay if they accidentally lose wifi.
